bugsnag-android-core/src/main/java/com/bugsnag/android/DefaultDelivery.kt

```diff
@@ -39,18 +44,53 @@ internal class DefaultDelivery(
 
     fun deliver(
         urlString: String,
-        streamable: JsonStream.Streamable,
+        eventPayload: EventPayload,
         headers: Map<String, String?>
     ): DeliveryStatus {
+        TrafficStats.setThreadStatsTag(1)
+        if (connectivity != null && !connectivity.hasNetworkConnection()) {
+            return DeliveryStatus.UNDELIVERED
+        }
 
+        var json = serializeJsonPayload(eventPayload)
+        if (json.size > maxPayloadSize) {
+            val breadcrumbAndBytesRemovedCounts =
+                eventPayload.event?.impl?.trimBreadcrumbsBy(json.size - maxPayloadSize)
```

Comment:
`eventPayload.event` will be `null` for unhandled errors, we'll need to load these from `eventFile`